src/controllers/routers/cases.js

```diff
@@ -74,14 +91,245 @@ function sanitizedCase(json) {
+      case 'beneficiaryname':
+        sortKey = 'beneficiary.name';
+        break;
+      case 'referencestatus':
+        sortKey = 'referenceStatus';
+        break;
+      case 'refereename':
+        sortKey = 'referee.name';
+        break;
+      case 'refereeorganisation':
+        sortKey = 'referee.organisation';
+        break;
+      case 'poc':
+        sortKey = 'poc';
+        break;
+      case 'appliedon':
+        sortKey = 'appliedOn';
+        break;
+      case 'updatedby':
+        sortKey = 'updatedAt';
+        break;
+      default:
+        sortKey = array[0].trim();
+    }
+
+    [query.sort_field, query.sort_order] = [sortKey, array[1].trim()];
+  }
+
+  // to check if applied_on is in YYYY-MM-DD format
+  if (
+    json.applied_on &&
+    /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/.test(
+      json.applied_on
+    ) === true
+  ) {
+    const year = json.applied_on.slice(0, 4);
+    const month = json.applied_on.slice(5, 7);
+    const day = json.applied_on.slice(8, 10);
+    query.applied_on = `${year}-${month}-${day}`;
+  }
+  return query;
+}
+
+/**
+ * Set default values based on query params.
+ * Call before an insert or an update.
+ * @param {Object} json - Query params without default values
+ * @return {Object} query - Query params with default values
+ */
+function setDefault(json) {
+  const query = json;
+  // if with_paging is true
+  if (json.with_paging === 'true') {
+    // if page is defined in query param, otherwise default to 1
+    if (!json.page || Number.isNaN(parseInt(json.page, 10)) === true) {
+      query.page = 1;
+    } else {
+      query.page = parseInt(json.page, 10);
+    }
+    // if per_page is defined in query param, otherwise default to 10
+    if (!json.per_page || Number.isNaN(parseInt(json.per_page, 10)) === true) {
+      query.per_page = 10;
+    } else {
+      query.per_page = parseInt(json.per_page, 10);
+    }
+  }
+  // if status is not present, set default to 'ALL'
+  if (!json.status) {
+    query.status = 'ALL';
+  }
+  // if sort is not present, set default to arrange id in asc
+  if (!json.sort) {
+    query.sort_field = 'id';
+    query.sort_order = 'asc';
+  }
+  return query;
+}
+
 /**
  * Retrieve all cases
  * @param {Request} req
  * @param {Response} res
  */
-const getAll = async (req, res) => {
-  const cases = await Case.query().select();
-  return res.status(200).json({ cases });
+const getAll = async (req, res, next) => {
+  const parsedQueries = sanitizedQuery(req.query);
+
+  // to set the default values for with_paging, page, per_page and status
+  setDefault(parsedQueries);
+
+  const currentPage = parsedQueries.page;
+  const limit = parsedQueries.per_page;
+  // if offset is undefined, set it to 0
+  // const offset = limit * currentPage - limit ? limit * currentPage - limit : 0;
+
+  try {
+    const requests = await Case.query()
+      .withGraphJoined(parsedQueries.include_entities)
+      .where((builder) => {
```

Comment:
the comparison should be case insensitive
I tested beneficiary_name and case number. 
When sending beneficiary_name == "a", the endpoint would return "Ziza" but not "Alex"
When sending case number that starts with  "ef", no cases are returned
Please apply case insensitive searches to rest of fields used in the search

Reply:
used 'ilike' for filtering relations (eg beneficiary_name, referee_name and referee_organisation) to account for case insensitive inputs

update sanitizedQuery to make case_number toUpperCase()